youtube_dl/extractor/vier.py

```diff
@@ -12,6 +14,101 @@
 )
 
 
+class VierVijfKijkOnlineIE(InfoExtractor):
+    IE_NAME = 'viervijfkijkonline'
+    IE_DESC = 'vier.be and vijf.be - Kijk Online'
+    _VALID_URL = r'https?://(?:www\.)?(?P<site>vier|vijf)\.be/video/(?P<series>(?!v3)[^/]+)/(?P<season>[^/]+)(/(?P<episode>[^/]+)|)'
+    _NETRC_MACHINE = 'vier'
+    _TESTS = [{
+        'url': 'https://www.vier.be/video/hotel-romantiek/2017/hotel-romantiek-aflevering-1',
+        'info_dict': {
+            'id': 'ebcd3c39-10a2-4730-b137-b0e7aaed247c',
+            'ext': 'mp4',
+            'title': 'Hotel Römantiek - Seizoen 1 - Aflevering 1',
+            'series': 'Hotel Römantiek',
+            'season_number': 1,
+            'episode_number': 1,
+        },
+        'skip': 'This video is only available for registered users'
+    }, {
+        'url': 'https://www.vier.be/video/blockbusters/in-juli-en-augustus-summer-classics',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.vier.be/video/achter-de-rug/2017/achter-de-rug-seizoen-1-aflevering-6',
+        'only_matching': True,
+    }]
+
+    def _real_initialize(self):
+        self._logged_in = False
+        self.id_token = ''
+
+    def _login(self):
+
+        username, password = self._get_login_info()
+        if username is None or password is None:
+            self.raise_login_required()
+
+        aws = AwsIdp(pool_id='eu-west-1_dViSsKM5Y', client_id='6s1h851s8uplco5h6mqh1jac8m')
+        self.id_token, _ = aws.authenticate(username=username, password=password)
+        self._logged_in = True
+
+    def _real_extract(self, url):
+
+        if not self._logged_in:
+            self._login()
+
+        webpage = self._download_webpage(url, None)
+
+        title = self._html_search_regex(
+            r'<meta\s*property="og:title"\s*content="(.+?)"\s*/>',
+            webpage, 'title')
+
+        title_split = title.split(' - ')
+        series = title_split[0].strip()
+        if len(title_split) == 3:
+            if 'Seizoen' in title_split[1]:
+                season = title_split[1].split('Seizoen')[1].strip()
+            else:
+                season = title_split[1].strip()
+            episode = title_split[2].split('Aflevering')[1].strip()
+        else:
+            season = None
+            episode = title_split[1].split('Aflevering')[1].strip()
```

Comment:
In case of a single episode (or a movie) this fails as there's no episode information in the title.

This can be tested with https://www.vier.be/video/trio/trio/trio-s1-aflevering-1 for example.

Reply:
Thanks for the feedback. I've just pushed a fix for this. At least, the example you mentioned now works.

Reply:
Those changes work correctly. 👍